tests/test_queryset.py

```diff
@@ -128,3 +130,39 @@ def test_validate():
 def test_map_synonyms():
     qs = ln.User.filter(handle="testuser1").all()
     assert qs.standardize(["user1", "user2"]) == ["user1", "user2"]
+
+
+def test_versioning():
+    # Create version 1 of artifact record
+    df_1 = pd.DataFrame(
+        {"CD8": [1, 2, 3], "CD45": [3, 4, 5]},
+        index=["measurement1", "measurement2", "measurement3"],
+    )
+    artifact_1 = ln.Artifact(df_1, description="Versioned test artifact")
+    artifact_1.save()
+
+    # Create version 2 of artifact record
+    df_2 = pd.DataFrame(
+        {"CD8": [4, 5, 6], "CD45": [7, 8, 9]},
+        index=["measurement1", "measurement2", "measurement3"],
+    )
+    artifact_2 = ln.Artifact(
+        df_2,
+        description="Versioned test artifact",
+        is_new_version_of=artifact_1,
+    )
+    artifact_2.save()
+
+    # Filter artifact by latest version
```

Comment:
How do we pass information about the version family here? 🤔 

Without characterizing the family, the "latest" query won't meet most users needs.

An easy way would be to pass `initial_version`, but the problem with this is that one has to know the initial version, which seems impractical.

Reply:
I think there are two user stories, here.

One is the one from the UI, or PyPI, or S3, or any other versioned experience where you don't whatever you're looking at be cluttered with all previous versions of the same thing.

The other one is to query for the latest version of a given version family, as in, "what's the latest version of this notebook".

Reply:
And damn it, the query would again be much simpler if we also labeled the seed element of the family with `initial_version`... it seems hairy and non-idiomatic to construct logic to fulfill user story #2 without it? Does it?

I mean, I think this here is something I can get used to
```
ln.Transform.filter(initial_version=my_reference_notebook.initial_version, version="latest")
```

But is this going to account for the seed element of the version family defined by `my_reference_notebook.initial_version`?

Reply:
Hmm, I see how this becomes hairy.

If you already have `my_reference_notebook`, couldn't you simply query for it:

```
my_notebook = ln.Transform.filter(id=my_reference_notebook.id).one()
```

And then list the versions via `.versions()`:

```
mynotebook_versions = my_notebook.versions()
```

And possibly also implement a `.latest_version()` property to have it be easily accessible?

I'm going to have to think about the straight versioned query. It's not immediately clear to me...

Reply:
The problem with this is that it takes two network requests and two lines of code:
<img width="681" alt="image" src="https://github.com/laminlabs/lamindb/assets/16916678/7447d01d-c214-4d4c-ac91-9daa0b32b0d9">

For something as fundamental and basic, I'd expect a single network request and one line of code.

Could you try the following out, which you can then also use for writing a test for this?
```
ln.Transform.filter(initial_version=my_reference_notebook.initial_version, version="latest")
```

In the worst case, we need to think about labeling the seed element of the version family with its own primary key.

Reply:
Agreed. Will try the query out and update my thinking based on your Slack arguments.